core/domain/collection_domain.py

```diff
@@ -100,78 +105,116 @@ class CollectionChange(change_domain.BaseChange):
         'required_attribute_names': ['first_index', 'second_index'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_EDIT_COLLECTION_PROPERTY,
         'required_attribute_names': ['property_name', 'new_value'],
         'optional_attribute_names': ['old_value'],
         'user_id_attribute_names': [],
-        'allowed_values': {'property_name': COLLECTION_PROPERTIES}
+        'allowed_values': {'property_name': COLLECTION_PROPERTIES},
+        'deprecated_values': {}
     }, {
         'name': CMD_EDIT_COLLECTION_NODE_PROPERTY,
         'required_attribute_names': [
             'exploration_id', 'property_name', 'new_value'],
         'optional_attribute_names': ['old_value'],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_MIGRATE_SCHEMA_TO_LATEST_VERSION,
         'required_attribute_names': ['from_version', 'to_version'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_ADD_COLLECTION_SKILL,
         'required_attribute_names': ['name'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_DELETE_COLLECTION_SKILL,
         'required_attribute_names': ['skill_id'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_ADD_QUESTION_ID_TO_SKILL,
         'required_attribute_names': ['question_id', 'skill_id'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_REMOVE_QUESTION_ID_FROM_SKILL,
         'required_attribute_names': ['question_id', 'skill_id'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }]
 
 
+class CollectionNodeDict(TypedDict):
+    """Dictionary representing the CollectionNode object."""
+
+    exploration_id: str
+
+
+class CollectionNodeVersionDict(CollectionNodeDict):
+    """Dictionary representing the CollectionNode object based
+    on different schema versions."""
+
+    acquired_skills: List[str]
+    acquired_skill_ids: List[str]
+    other_field: str
+    prerequisite_skills: List[str]
+    prerequisite_skill_ids: List[str]
```

Comment:
Why we need two of these?

Reply:
As I said in other places lets ignore all errors in `_convert_*` and only create the latest version of the `TypedDict`